sklearn/utils/validation.py

```diff
@@ -1084,6 +1084,13 @@ def is_sparse(dtype):
                 % (n_features, array.shape, ensure_min_features, context)
             )
 
+    # With pandas copy-on-write and if copy=True we know we can write to the
+    # pandas dataframe or series, see
+    # https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html#read-only-numpy-arrays
+    # for more details about pandas copy-on-write
+    if _is_pandas_df_or_series(array_orig) and copy:
```

Comment:
```suggestion
    if _is_pandas_df_or_series(array_orig):
```

If the user has set `copy=False` explicitly in the estimator, we also have his/her permission to write to the original dataframe as well via the numpy view.